include/dali/core/error_handling.h

```diff
@@ -234,28 +248,28 @@ inline dali::string GetStacktrace() {
 #define FILE_AND_LINE __FILE__ ":" DALI_STR(__LINE__)
 
 #define DALI_MESSAGE_WITH_STACKTRACE(str)\
-  (std::string("[" FILE_AND_LINE "] ") + str + dali::GetStacktrace())
+  (std::string("[" FILE_AND_LINE "] ")), (std::string(str)), dali::GetStacktrace()
 
 #define DALI_MESSAGE(str)\
-  (std::string("[" FILE_AND_LINE "] ") + str)
+  (std::string("[" FILE_AND_LINE "] ")), (std::string(str))
 
 #define DALI_FAIL(str)                            \
     throw dali::DALIException(DALI_MESSAGE_WITH_STACKTRACE(str));
 
-#define DALI_ERROR(str)                                          \
-  do {                                                           \
-    std::cerr << DALI_MESSAGE_WITH_STACKTRACE(str) << std::endl; \
+#define DALI_ERROR(str)                                                             \
+  do {                                                                              \
+    std::cerr << dali::make_string(DALI_MESSAGE_WITH_STACKTRACE(str)) << std::endl; \
   } while (0)
 
-#define DALI_WARN(...)                                                      \
-  do {                                                                      \
-    std::cerr << DALI_MESSAGE(dali::make_string(__VA_ARGS__)) << std::endl; \
+#define DALI_WARN(str)                                              \
+  do {                                                              \
+    std::cerr << dali::make_string(DALI_MESSAGE(str)) << std::endl; \
```

Comment:
We should probably have some synchronized print... Stringification before printing helps a big, but not much.
```suggestion
    diag_print(DALI_MESSAGE(str));
```
where `diag_print` has a mutex.

Reply:
I will consider this out of scope for now.